components/movementsensor/gpsrtkpmtk/gpsrtkpmtk_test.go

```diff
@@ -0,0 +1,189 @@
+package gpsrtkpmtk
+
+import (
+	"context"
+	"math"
+	"testing"
+
+	"github.com/edaniels/golog"
+	geo "github.com/kellydunn/golang-geo"
+	"go.viam.com/test"
+	"go.viam.com/utils"
+
+	"go.viam.com/rdk/components/movementsensor/fake"
+	rtk "go.viam.com/rdk/components/movementsensor/rtkutils"
+)
+
+var (
+	alt   = 50.5
+	speed = 5.4
+)
+
+const (
+	testRoverName   = "testRover"
+	testStationName = "testStation"
+	testBoardName   = "board1"
+	testBusName     = "bus1"
+	testi2cAddr     = 44
+)
+
+func TestValidateRTK(t *testing.T) {
+	path := "path"
+	fakecfg := &Config{
+		Board:                testBoardName,
+		I2CBus:               testBusName,
+		I2CAddr:              testi2cAddr,
+		I2CBaudRate:          4400,
+		NtripURL:             "fakeurl",
+		NtripConnectAttempts: 10,
+		NtripPass:            "somepass",
+		NtripUser:            "someuser",
+		NtripMountpoint:      "NYC",
+	}
+
+	_, err := fakecfg.Validate(path)
+	test.That(t, err, test.ShouldBeNil)
+
+	fakecfg.NtripURL = ""
+	_, err = fakecfg.Validate(path)
+	test.That(t, err, test.ShouldBeError, utils.NewConfigValidationFieldRequiredError(path, "ntrip_url"))
+
+	fakecfg.NtripURL = "http://fakeurl"
+	fakecfg.I2CBus = ""
+	_, err = fakecfg.Validate(path)
+	test.That(
+		t,
+		err,
+		test.ShouldBeError,
+		utils.NewConfigValidationFieldRequiredError(path, "i2c_bus"),
+	)
+	fakecfg.I2CBus = testBusName
+	fakecfg.I2CAddr = 0
+	_, err = fakecfg.Validate("path")
+	test.That(
+		t,
+		err,
+		test.ShouldBeError,
+		utils.NewConfigValidationFieldRequiredError(path, "i2c_addr"))
+}
+
+func TestConnect(t *testing.T) {
+	logger := golog.NewTestLogger(t)
+	ctx := context.Background()
+	cancelCtx, cancelFunc := context.WithCancel(ctx)
+	g := rtkI2C{
+		cancelCtx:  cancelCtx,
+		cancelFunc: cancelFunc,
+		logger:     logger,
+	}
+
+	url := "http://fakeurl"
+	username := "user"
+	password := "pwd"
+
+	// create new ntrip client and connect
+	err := g.connect("invalidurl", username, password, 10)
+	g.ntripClient = makeMockNtripClient()
+
+	test.That(t, err, test.ShouldNotBeNil)
+
+	err = g.connect(url, username, password, 10)
+	test.That(t, err, test.ShouldBeNil)
+
+	err = g.getStream("", 10)
```

Comment:
what is this testing exactly? Checking for the specific error returned in the scenario you're testing could also make this clearer